### ISSUE TITLE
Recreating controls often throws errors

### ISSUE THREAD

**Issue by JeroenVinke** [aurelia-ui-toolkits/aurelia-kendoui-bridge#482]
https://gist.run/?id=2e2b368306242ea500310bd675a39c53 

```
        this.gridVM.kWidget.destroy();
        kendo.destroy(this.gridVM.kWidget);
        $(this.test).empty();
        this.gridVM.recreate();
```

is needed to destroy and recreate a grid... we need to improve this

**Comment by adriatic** [comment]
This is convoluted indeed. Let me see if I understand - using the "bridge side" and "kendo side" classification.

`this.gridVM.kWidget.destroy();` reaches into "kendo side" to destroy the "real object"
`kendo.destroy(this.gridVM.kWidget);` destroys the bridge side reference to just destroyed real object.

Not sure what is `$(this.test).empty();` - anything to do with continuous testing?

Finally - what context is still present for `this.gridVM.recreate()` to make sense? I guess that everything in VM was un-affected with cleanup done so far?

**Comment by JeroenVinke** [comment]
I need to look into this further, but I believe that 
`this.gridVM.kWidget.destroy();`
destroys the widget

and 
`kendo.destroy(this.gridVM.kWidget);`
destroys the widget and any child widgets within

`$(this.test).empty();` - the grid is initialzed on the `this.test` div, so this line clears up any HTML that's still in the div. Somehow Kendo doesn't clean up the HTML, also something we should look into

**Comment by adriatic** [comment]
> I need to look into this further, but I believe that 
`this.gridVM.kWidget.destroy();`
destroys the widget

>and 
`kendo.destroy(this.gridVM.kWidget);`
destroys the widget and any child widgets within

This would indicate that one should use **either** `this.gridVM.kWidget.destroy();` **or** `kendo.destroy(this.gridVM.kWidget);` - **but not both**

**Comment by JeroenVinke** [comment]
yep, good one. I just looked at http://docs.telerik.com/kendo-ui/intro/widget-basics/destroy and it mentions this:

> Creating a new widget instance from the leftovers of a destroyed widget may work, but is not recommended. Initialize new widgets from different, such as newly appended, DOM elements. In some cases it is also possible to empty the widget container and initialize a new instance from the empty element.

Unfortunately, we sometimes initialize kendo controls on the custom element itself, not always on a div/span/input, so I don't think it's possible to "clone" a custom element. Perhaps we should always initialize kendo controls on a span/div/input so we can remove it and create a new one.

**Comment by adriatic** [comment]
> Perhaps we should always initialize kendo controls on a span/div/input ...

I would think that this is a very good reason to change what we have today.
